### DISCUSSION TITLE
Make "alias" an actual alias of a data property and change "alias" to "metavar"

### DISCUSSION THREAD

**Discussion by philipptempel** [pydantic/pydantic#2359] [Feature Request]
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this feature/change is needed
* [x] After submitting this, I commit to one of:
    * Look through open issues and helped at least one other person
    * Hit the "watch" button on this repo to receive notifications and I commit to help at least 2 people that ask questions in the future
    * Implement a Pull Request for a confirmed bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.7.3
            pydantic compiled: True
                 install path: ~/Library/Caches/pypoetry/virtualenvs/pyrefmate--35n53Wu-py3.7/lib/python3.7/site-packages/pydantic
               python version: 3.7.9 (default, Sep  6 2020, 13:20:25)  [Clang 11.0.3 (clang-1103.0.32.62)]
                     platform: Darwin-19.6.0-x86_64-i386-64bit
     optional deps. installed: ['typing-extensions', 'devtools']
```

(additional information / motivation at the bottom)

# Question (or Feature Request?)

## Question
Is it by design that `bar = Field(alias="baz")` is not really an alias property `baz` of the data in `bar` but a "name substitute"? And how can I then crate real aliases for my data such that one datum is accessible (applies more to creating objects rather than manipulating objects) through two different properties?

## Feature Request
Replace `alias` in `Field` with `metavar` and make `alias` actually create real aliases.

I got very confused with the section on `aliases` in pydantic since--I don't want to be too pedantic here ;) --an alias 
> an assumed or additional name that a person (such as a criminal) sometimes uses".

According to Wikipedia, ["Alias (Computing)"](https://en.wikipedia.org/wiki/Aliasing_(computing)) is
> "In computing, aliasing describes a situation in which a data location in memory can be accessed through different symbolic names in the program"

Thus, I assumed when specifying `alias="something"` during declaration of a field, that there would now be two fields in my object: `something` being the alias for my other property and both pointing to the same data.

As I have been developing several python CLI scripts lately (using packages like `click` and `typer`), it seems more natural to me to use `metavar` to describe the publicly visible property and use a meaningful property name for internal use.
Reading about `alias` in `pydantic`, I was thinking that this way, I could have two attributes which are internally linked to the same data. Like in my example below, a `Person` has one `location`, but it may be publicly visible as both an `address` and a `location`.

Self-contained code snippet of current code with expected behavior
```py
from __future__ import  annotations
import pydantic
from devtools import debug


class Person(pydantic.BaseModel):
    name: str
    location: str = pydantic.Field(alias="address")


def main():
    # create with original attribute: fails (address field required (type=value_error.missing)
    john = Person(name="John Doe", location="123 East Main Street")
    debug(john)
    debug(john.address)
    debug(john.location)

    # create with aliased attribute
    jane = Person(name="Jane Doe", address="123 East Main Street")
    debug(jane)
    debug(jane.location)
    debug(jane.address) # AttributeError: 'Person' object has no attribute 'address'


if __name__ == "__main__":
    main()
```

A vanilla code version of my intended behavior would be
```py
from devtools import debug


class Person(object):

    def __init__(self, *_, name: str = None, location: str = None, address: str = None):
        self.name = name if name else ""
        self._location_or_address = location if location else ( address if address else "")

    @property
    def location(self):
        return self._location_or_address
    @location.setter
    def location(self, l: str):
        self._location_or_address = l

    @property
    def address(self):
        return self._location_or_address
    @address.setter
    def address(self, a: str):
        self._location_or_address = a



def main():
    # create with original attribute
    john = Person(name="John Doe", location="123 East Main Street")
    debug(john)
    debug(john.address)
    debug(john.location)

    # create with aliased attribute
    jane = Person(name="Jane Doe", address="123 East Main Street")
    debug(jane)
    debug(jane.location)
    debug(jane.address)


if __name__ == "__main__":
    main()

```

I am particularly interested in this FR as I think it would give developers more freedom to use pydantic models to also parse structured data where there may be a field `location` in one datum and a field `address` in the next datum, despite them actually being the same content.

Additional information / motivation:
The TUI I am currently developing is a reference management CLI which should be able to read BibTeX and BibLaTeX files, where BibTeX-files declare an `address` and BibLaTeX-files declare a `location` for each entry. The contained datum is the same, but the key is a different. So a property _alias_ is what I (think I) am looking for.

Discussion for issue #2244

**Comment by samuelcolvin** [comment]
Have you looked at [`allow_population_by_field_name`](https://pydantic-docs.helpmanual.io/usage/model_config/). There's also `by_alias` on `model.dict()`.

I'm not changing "alias" to "metavar", it is no more descriptive.
